docusaurus/docs/how-to-guides/ui-extensions/register-an-extension.md

```diff
@@ -0,0 +1,232 @@
+---
+id: register-an-extension
+title: Register an extension
+description: Make part of the UI extendable with links or components defined by plugins
+keywords:
+  - grafana
+  - plugins
+  - plugin
+  - extensions
+  - ui-extensions
+sidebar_position: 20
+---
+
+An "extension point" is a part of your plugin UI or Grafana UI where other plugins can hook in with either links or React components to extend the user experience. [Read more about extensions under key concepts](../../key-concepts/ui-extensions.md).
+| Type | Description |
+| ------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| **Link** | Links have a `path` and an `onClick()` property. <br /><br /> **When to use?** <br /> Use links if you would like to give plugins a way to define custom user actions for a part of your UI. These actions can either just be cross-links to the plugin, or using `onClick()` methods they can implement a more interactive on-page experience with a modal. <br /><br /> **API reference** <br /> - [`addLink()`](../../reference/ui-extensions.md#addlink) - registering a link from a plugin <br /> - [`usePluginLinks()`](../../reference/ui-extensions.md#usepluginlinks) - fetching links registered for an extension point |
+| **Component** | Components are React components that can be used to render a custom user experience. <br /><br /> **When to use?** <br /> Use components if you would like to give more freedom for plugins to extend your UI, for example to extend a configuration form with custom parts. <br /><br /> **API reference** <br /> - [`addComponent()`](../../reference/ui-extensions.md#addcomponent) - registering a component from a plugin <br /> - [`usePluginComponents()`](../../reference/ui-extensions.md#useplugincomponents) - fetching components registered for an extension point |
+
+## Links
+
+### Best practices for adding links
+
+- **Use the `configure()` function** - if your link's path has variables in it that are coming from the context, make sure to use the `configure()` function for updating the path.
```

Comment:
I'm struggling with this as a best practice, just "Use the configure function"

Maybe the sentiment is something like:
`Make your path dynamic - if appropriate, you can alter your path based on the context and provide a more targeted link. See [Update the path based on the context](link)`

Reply:
removing this bullet point entirely

Reply:
Yeah, feels like a "tips and tricks". I guess we also could mention that if you return undefined from the configure function the link will be hidden.